Added InsertBoundary and Select from level

Feuerbowle opened the pull request:
Functions added to insert one boundary into one other.
- InsertBoundary(PMBoundary): Insert a boundary entity
- InsertBoundaryByName(String): Insert a boundary by name

Functions added to select all/wireframe/surfeces from an level or set
- SelectAll(): Select all from that level or set
- SelectWireframe(): Select all wireframes from that level or set
- SelectSurfaces(): Select all surfaces from that level or set

Feuerbowle committed:
Added InsertBoundary and Select from level

Functions added to insert one boundary into one other.
- InsertBoundary(PMBoundary): Insert a boundary entity
- InsertBoundaryByName(String): Insert a boundary by name

Functions added to select all/wireframe/surfeces from an level or set
- SelectAll(): Select all from that level or set
- SelectWireframe(): Select all wireframes from that level or set
- SelectSurfaces(): Select all surfaces from that level or set

lukeedw commented on Delcam.ProductInterface.PowerMILL/Classes/PMBoundary.cs:
```diff
@@ -135,6 +135,24 @@ public void Smash(double smashTolerance)
+        /// <param name="boundary">The boundary to insert into this Boundary.</param>
+        public void InsertBoundary(PMBoundary boundary)
+        {
+            PowerMill.DoCommand("EDIT BOUNDARY '" + Name + "' INSERT BOUNDARY '" + boundary.Name + "'");
```
Is it worth testing that the boundary is not null and that it exists in PowerMill?

lukeedw commented on Delcam.ProductInterface.PowerMILL/Classes/PMBoundary.cs:
```diff
@@ -135,6 +135,24 @@ public void Smash(double smashTolerance)
+        /// <param name="boundaryName">The name of the boundary to insert into this Boundary.</param>
+        public void InsertBoundaryByName(string boundaryName)
+        {
+            PowerMill.DoCommand("EDIT BOUNDARY '" + Name + "' INSERT BOUNDARY '" + PowerMill.ActiveProject.Boundaries.GetByName(boundaryName).Name + "'");
```
Same here.  Test that the boundary is not null?

lukeedw commented on Delcam.ProductInterface.PowerMILL/Classes/PMLevelOrSet.cs:
```diff
@@ -88,6 +88,33 @@ public override void Delete()
+        /// <remarks></remarks>
+        public void SelectAll()
+        {
+            PowerMill.DoCommand("EDIT LEVEL '" + this.Name + "' SELECT ALL");
```
No need to use "this"

lukeedw commented on Delcam.ProductInterface.PowerMILL/Classes/PMLevelOrSet.cs:
```diff
@@ -88,6 +88,33 @@ public override void Delete()
+        /// <remarks></remarks>
+        public void SelectWireframe()
+        {
+            PowerMill.DoCommand("EDIT LEVEL '" + this.Name + "' SELECT WIREFRAME");
```
No need to use "this"

lukeedw commented on Delcam.ProductInterface.PowerMILL/Classes/PMLevelOrSet.cs:
```diff
@@ -88,6 +88,33 @@ public override void Delete()
+        /// <remarks></remarks>
+        public void SelectSurfaces()
+        {
+            PowerMill.DoCommand("EDIT LEVEL '" + this.Name + "' SELECT SURFACE");
```
No need to use "this"

lukeedw reviewed (commented):
Made a few comments but looks great.

Feuerbowle committed:
Minor changes

PMLevelOrSets
- changes "this.Name" to "Name"

PMBoundary
- Check if boundary is null and does exists
